### PR TITLE
Add Support to RTL on View Pager 

### PR DESCRIPTION
**Pull request by wching**
## What I did

Added an attribute named

`IndefinitePagerIndicator_supportRTL`

on the View that will allow the users to support RTL on Custom View Pagers. It default's on **false** to avoid conflicts.

Basically the feature just inverts the current position when the View **is** RTL and the flag is on, this to give the user the flexibility to decide even when RTL being used to support it or not.

## Sample Information

Added the following library

https://github.com/duolingo/rtl-viewpager as the native support viewpager doesn't support rtl out of the box. This is a custom implementation and it works just fine with **Android-Indefinite-Pager-Indicator** 🌮 🌮

### CODE REVIEW

File: indefinitepagerindicator/src/main/java/com/rbrooks/indefinitepagerindicator/IndefinitePagerIndicator.kt [hunk 1]
```diff
@@ -39,11 +41,13 @@ class IndefinitePagerIndicator @JvmOverloads constructor(context: Context, attrs
     private var selectedDotRadiusPx = dpToPx(DEFAULT_SELECTED_DOT_RADIUS_DP, resources)
     private var dotRadiusPx = dpToPx(DEFAULT_DOT_RADIUS_DP.toFloat(), resources)
     private var dotSeparationDistancePx = dpToPx(DEFAULT_DOT_SEPARATION_DISTANCE_DP.toFloat(), resources)
+    private var supportRtl = false
 
     @ColorInt
-    private var dotColor: Int = resources.getColor(R.color.default_dot_color)
+    private var dotColor: Int = ContextCompat.getColor(this.context, R.color.default_dot_color)
```
Comment (rbro112) [inline indefinitepagerindicator/src/main/java/com/rbrooks/indefinitepagerindicator/IndefinitePagerIndicator.kt:47]: Nice catch, thanks!

File: sample/src/main/res/layout/rtl_fragment_view_pager_sample.xml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        android:background="@color/white"
+        android:elevation="4dp"
+        android:orientation="horizontal">
+
+        <Button
+            android:id="@+id/viewpager_previous_button"
+            style="@style/PrimaryFlatButton"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_alignParentLeft="true"
```
Comment (rbro112) [inline sample/src/main/res/layout/rtl_fragment_view_pager_sample.xml]: Would you mind adding the ```layout_alignParentLeft="true"``` attribute to the ```viewpager_previous_button``` in ```fragment_view_pager_sample.xml```? The bottom nav bar doesn't render correctly in the normal ViewPager sample when in RTL layout.

File: sample/src/main/res/layout/rtl_fragment_view_pager_sample.xml [hunk 3]
```diff
@@ -0,0 +1,61 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (rbro112) [inline sample/src/main/res/layout/rtl_fragment_view_pager_sample.xml]: Could we rename this file to ```fragment_rtl_view_pager_sample.xml```? A slight nitpick, I just like to have fragment_/activity_/viewholder_/etc as my prefix for xml files.

**Comment by rbro112** [comment]
This looks great, minus some quick comments! Thanks for adding this. Once the comments are addressed and I can test again, I'll merge and update the library to v1.0.5 on Maven including these changes.

File: sample/src/main/res/layout/rtl_fragment_view_pager_sample.xml (hunk 2 above)
Comment (wching) [reply sample/src/main/res/layout/rtl_fragment_view_pager_sample.xml]: Sure thing! 🌮

File: sample/src/main/res/layout/rtl_fragment_view_pager_sample.xml (hunk 3 above)
Comment (wching) [reply sample/src/main/res/layout/rtl_fragment_view_pager_sample.xml]: sure thing! 🌮

File: indefinitepagerindicator/src/main/java/com/rbrooks/indefinitepagerindicator/IndefinitePagerIndicator.kt (hunk 1 above)
Comment (wching) [reply indefinitepagerindicator/src/main/java/com/rbrooks/indefinitepagerindicator/IndefinitePagerIndicator.kt:47]: You're welcome! 👍

**Comment by wching** [comment]
@rbro112 changes addressed! Thanks so much for the quick review! 🌮 🌯

**Comment by rbro112** [comment]
Works awesome! Merging now. I'll put a PR up with v1.0.5 updates shortly and tag you in it.

**Pull request merged**